Library/EnvelopeFactory.ts

```diff
@@ -1,13 +1,163 @@
+        const document = EnvelopeFactory.createQuickPulseDocument(envelope);
+
+        var traceDocument: Contracts.MessageDocumentQuickPulse = {
+            ...document,
+            Properties: [{key: 'my key', value: 'my value'}],
+            Message: ((envelope.data as any).baseData as Contracts.MessageData).message,
+            SeverityLevel: "Critical"
+        }
+
+        return traceDocument;
+    }
+
+    private static createQuickPulseExceptionDocument(envelope: Contracts.Envelope): Contracts.ExceptionDocumentQuickPulse {
+        const document = EnvelopeFactory.createQuickPulseDocument(envelope);
+
+        let exception = '';
+        ((envelope.data as any).baseData as Contracts.ExceptionData).exceptions[0].parsedStack.forEach(err => {
+            exception += err.assembly + "\n";
+        })
+        var exceptionDocument = {
+            ...document,
+            Exception: exception,
+            ExceptionMessage: ((envelope.data as any).baseData as Contracts.ExceptionData).exceptions[0].message,
+            ExceptionType: ((envelope.data as any).baseData as Contracts.ExceptionData).exceptions[0].typeName
+        };
+        return exceptionDocument;
+    }
+
+    private static createQuickPulseRequestDocument(envelope: Contracts.Envelope): Contracts.RequestDocumentQuickPulse {
+        const document = EnvelopeFactory.createQuickPulseDocument(envelope);
+        const baseData = (envelope.data as Contracts.Data<Contracts.RequestData>).baseData;
+        const requestDocument: Contracts.RequestDocumentQuickPulse = {
+            ...document,
+            Name: baseData.name,
+            Success: baseData.success,
+            Duration: baseData.duration,
+            ResponseCode: baseData.responseCode,
+            OperationName: baseData.name // TODO: is this correct?
+        };
+
+        return requestDocument;
+    }
+
+    private static createQuickPulseDependencyDocument(envelope: Contracts.Envelope): Contracts.DependencyDocumentQuickPulse {
+        const document = EnvelopeFactory.createQuickPulseDocument(envelope);
+        const baseData = (envelope.data as Contracts.Data<Contracts.RemoteDependencyData>).baseData;
+
+        const dependencyDocument: Contracts.DependencyDocumentQuickPulse = {
+            ...document,
+            Name: baseData.name,
+            Target: baseData.target,
+            Success: baseData.success,
+            Duration: baseData.duration,
+            ResultCode: baseData.resultCode,
+            CommandName: baseData.data,
+            OperationName: document.OperationId,
+            DependencyTypeName: baseData.type,
+        }
+        return dependencyDocument;
+    }
+
+    private static createQuickPulseDocument(envelope: Contracts.Envelope): Contracts.DocumentQuickPulse {
+        let documentType, __type, id, operationId;
+
+        switch (envelope.data.baseType) {
+            case "EventData":
```

Comment:
We might want to consider enums for all these constants

Reply:
updated to add constants and mappings for all of these. default EnvelopeFactory is using numbered enums for these fields, so I created new `TelemetryTypeString` enum for this next to existing one.